staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -402,7 +406,16 @@ func (s *store) GetToList(ctx context.Context, key string, listOpts storage.List
 
 	key = path.Join(s.pathPrefix, key)
 	startTime := time.Now()
-	getResp, err := s.client.KV.Get(ctx, key, s.getOps...)
+	opts := s.getOps
+	if len(resourceVersion) > 0 && match == metav1.ResourceVersionMatchExact {
+		rv, err := s.versioner.ParseResourceVersion(resourceVersion)
+		if err != nil {
+			return apierrors.NewBadRequest(fmt.Sprintf("invalid resource version: %v", err))
+		}
+		opts = append(opts, clientv3.WithRev(int64(rv)))
+	}
+
+	getResp, err := s.client.KV.Get(ctx, key, opts...)
 	metrics.RecordEtcdRequestLatency("get", getTypeName(listPtr), startTime)
 	if err != nil {
 		return err
```

Comment:
just below, is the call to ensureMinimumResourceVersion is only needed if `match == metav1.ResourceVersionMatchNotOlderThan || match == ""`?

Reply:
Calling it for ExactMatch doesn't break anything, because it should be equal then (so also <=). Just additional safety check.

Reply:
Yeah, it looks like we can always run the check.

To be consistent about it, if we're always going to do the check here, then I should also always do it in `List` and `GetToList`. I've updated the code.